docs/proposals/kep-002-bulward.md

```diff
@@ -0,0 +1,107 @@
+# [KEP-002] Bulward integration
+
+## Background
+
+Currently, we have `Account` concept in kubecarrier, which encompasses single user. Due to bigger interop with the Kubermatic ecosystem, as well as further requirements on the user management we built `bulward`. The account is permeated through the kubecarrier project, as it's central user concept. The API server's proto file specify the account, the accounts come into two possible roles (Provider and Tenant), and adequate RBAC permissions
+
+## Migration plan
+
+Current `Account` concept closely matches bulward's `Project`. It's a sandbox where single focused responsibility hosts its resources. Thus, all references to the account through the kubecarrier solution should be renamed to account. In the following code excerpt you can see similarities and differences.
+
+```yaml
+apiVersion: catalog.kubecarrier.io/v1alpha1
+kind: Account
+metadata:
+  name: team-a
+spec:
+  metadata:
+    displayName: The A Team
+    shortDescription: In 1972, a crack commando unit was sent to prison by a military court...
+  roles:
+  - Provider
+  subjects:
+  - kind: User
+    name: hannibal
+    apiGroup: rbac.authorization.k8s.io
+  - kind: User
+    name: team-a-member
+    apiGroup: rbac.authorization.k8s.io
+```
+
+```yaml
+apiVersion: apiserver.bulward.io/v1alpha1
+kind: Project
+metadata:
+  name: project-example
+  namespace: organization-a
```

Comment:
Do we require the Organization to be created before creating Project? Because Project objects are namespaced-scoped, we should think about if we force to create Organization first

Reply:
that's a bulward thing...I'd say yes. Even if you have a single root organization.

Reply:
btw, we now have defined arbitrary hierarchy two levels deep without recursion (projects are part of the organization). We currently don't support recursion (i.e. projects cannot contain multiple sub-project) nor deeper hierarchies.